app/models/manageiq/providers/azure/cloud_manager/provision/cloning.rb

```diff
@@ -0,0 +1,143 @@
+module ManageIQ::Providers::Azure::CloudManager::Provision::Cloning
+  def do_clone_task_check(clone_task_ref)
+    source.with_provider_connection do |azure|
+      vms               = Azure::Armrest::VirtualMachineService.new(azure)
+      vm_name           = clone_task_ref[3]
+      vm_resource_group = clone_task_ref[1]
```

Comment:
`clone_task_ref` is the value returned from the `start_clone` at the bottom of this file.  Usually it is just an ID, but if you need to return more items a hash would be a better choice so you could provide meaningful names to the data elements.